src/Native/LibMFWrapper/mf.cpp

```diff
@@ -0,0 +1,1314 @@
+/*
+Copyright(c) 2014 - 2015 The LIBMF Project.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met :
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and / or other materials provided with the distribution.
+
+3. Neither name of copyright holders nor the names of its contributors
+may be used to endorse or promote products derived from this software
+without specific prior written permission.
+
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED.IN NO EVENT SHALL THE REGENTS OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT(INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+#include <algorithm>
+#include <condition_variable>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <queue>
+#include <thread>
+#include <unordered_set>
+#include <random>
+#include <cstring>
+#include <fstream>
+#include <vector>
+#include <new>
+#include <string>
+#include <memory>
+
+#include "mf.h"
+
+#if defined USESSE
+#include <pmmintrin.h>
+#endif
+
+#if defined USEAVX
+#include <immintrin.h>
+#endif
+
+#if defined USEOMP
+#include <omp.h>
+#endif
+
```

Comment:
Do we actually define any of these? And do we plan to, because if not, you can remove a lot of code.

Reply:
`mf.cpp` and `mf.h` were copied from the external package, LibMF, so I suggest not to touch them.

Reply:
LIBMF is a submodule now, so I feel we can leave it as it.

---
In reply to: [225361232](https://github.com/dotnet/machinelearning/pull/1263#discussion_r225361232) [](ancestors = 225361232)